lib/src/main/java/io/ably/lib/realtime/AblyRealtime.java

```diff
@@ -71,6 +71,13 @@ public void onAuthUpdated() {
 	}
 
 	/**
+	 * Authentication error occurred
+	 */
+	public void onAuthError(ErrorInfo errorInfo) {
```

Comment:
Matt:
> In fact, whatever calls onAuthError could instead just call 
> connection.connectionManager.onAuthError surely

No, because onAuthError is called from rest.Auth, which doesn't know anything about realtime and connections.

Paddy:
> But isn't this only called from rest.Auth ? And that is calling the method declared on 
> rest.AblyRest ?

Yes, this might work. Make AblyRest.onAuthError protected so (a) it cannot be called from outside the rest package, but (b) it can be overridden by AblyRealtime, which is a subclass of AblyRest.

The same thing applies to onAuthUpdated that I added in the in-place auth changes. It too should be protected.